File: Beginner/schoolGeo/schoolGeo.java
//run as java -ea schoolGeo
import java.util.Scanner;
import java.util.Arrays;

public class schoolGeo{
	public static void main(String[] args){
		int T,N,sum;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=50) : "T must be 1 <= T <= 50";

		while(T>0){
			N = scan.nextInt();
			assert(N>=1 && N<=10000) : "N must be 1 <= N <= 10^4";

			int[] A = new int[N];
			int[] B = new int[N];

			for(int i=0;i<N;i++){
				A[i] = scan.nextInt();
				assert(A[i]>=1 && A[i]<=1000000000) : "A["+i+"] must be 1 <= A[i] <= 10^9";
			}

			for(int i=0;i<N;i++){
				B[i] = scan.nextInt();
				assert(B[i]>=1 && B[i]<=1000000000) : "B["+i+"] must be 1 <= B[i] <= 10^9";
			}

			Arrays.sort(A);
			Arrays.sort(B);

			sum = 0;

			for(int i=0;i<N;i++){
				sum += min(A[i],B[i]);
			}

			System.out.println(sum);

			T--;
		}
	}

	public static int min(int a,int b){
		if(a<b){
			return a;
		}else{
			return b;
		}
	}
}

File: Beginner/schoolGeo/schoolGeo.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>

int compare(const void *a,const void *b){
	return *(int*)a-*(int*)b;
}

int min(int a,int b){
	if(a<b){
		return a;
	}else{
		return b;
	}
}

int main(void){
	int T,N,sum;

	scanf("%d",&T);
	assert(T>=1 && T<=50);

	while(T>0){
		scanf("%d",&N);
		assert(N>=1 && N<=10000);

		int A[N];
		int B[N];

		for(int i=0;i<N;i++){
			scanf("%d",&A[i]);
			assert(A[i]>=1 && A[i]<=1000000000);
		}

		for(int i=0;i<N;i++){
			scanf("%d",&B[i]);
			assert(B[i]>=1 && B[i]<=1000000000);
		}

		qsort(A,N,sizeof(int),compare);
		qsort(B,N,sizeof(int),compare);

		sum = 0;

		for(int i=0;i<N;i++){
			sum += min(A[i],B[i]);
		}

		printf("%d\n",sum);

		T--;
	}

	return 0;
}